packages/core/src/tools/create-file-stasher.js

```diff
@@ -237,6 +235,19 @@ const uploader = async (
   throw new Error(`Got ${response.status} - ${response.content}`);
 };
 
+const isUploadMaxSizeExceeded = (streamOrData, length) => {
+  let uploadMaxSize = NON_STREAM_UPLOAD_MAX_SIZE;
+  let uploadMethod = 'non-streaming';
+  if (streamOrData instanceof Readable) {
+    uploadMaxSize = UPLOAD_MAX_SIZE;
+    uploadMethod = 'streaming';
+  }
+
+  if (length && length > uploadMaxSize) {
+    throw new Error(`${length} is too big, ${uploadMaxSize} is the max for ${uploadMethod} of data.`);
```

Comment:
That "of" sounds redundant. We might want to add the size unit too so it's more clear.

```suggestion
    throw new Error(`${length} bytes is too big, ${uploadMaxSize} is the max for ${uploadMethod} data.`);
```

Reply:
Done